### PR TITLE
Gemini ai intergration

### PR DESCRIPTION
**Pull request by millyanne93**
## Description

This PR transitions our AI question classification and recommendation system from PyTorch to Gemini AI. The key changes and reasons for the transition are outlined below.

## Key Changes

1. **Updated Dependencies:**
- Removed PyTorch and related dependencies.
- Added the @google/generative-ai package.

2. **New Gemini AI Integration:**
- Created a new file `gemini_model.js` in the `ai/` directory for interfacing with the Gemini AI API.
   - Refactored the file to use `GoogleGenerativeAI` class from the `@google/generative-ai` package.
   - Updated the methods for generating content and processing questions to match the new API requirements.

3. **Modified Upload Controller:**
- Updated `uploadController.js` to call the new Gemini AI-based classification and recommendation functions.

4. **Configuration Changes:**
- Updated `.env` file to include `GEMINI_API_KEY`.
- Modified `package.json` to reflect the new dependencies and scripts.

## Why Gemini AI?

- **Improved Accuracy:** Gemini AI offers state-of-the-art performance for natural language understanding and generation tasks.
- **Scalability:** Gemini AI is cloud-based, offering better scalability and maintenance compared to our previous setup.
- **Simplified Codebase:** Using Gemini AI simplifies our codebase by offloading the heavy lifting to a managed service.

### CODE REVIEW

File: backend/data/word2idx.json
Comment (ynot93) [inline backend/data/word2idx.json:1]: This file was to be removed seeing as we are not using it

File: backend/utils/train_model.py
Comment (ynot93) [inline backend/utils/train_model.py:1]: Are we using this file too?

**Review by ynot93 — CHANGES_REQUESTED** [review]
Looks good to me. Check on whether we need the files I've commented on

File: backend/utils/train_model.py
Comment (millyanne93) [reply backend/utils/train_model.py:1]: I had deleted them but i realized i did not commit the changes on my local environment . let me do that ASAP

**Pull request merged**